app/services/supplementary_data.py

```diff
@@ -90,3 +124,8 @@ def validate_supplementary_data(
         )
     except ValidationError as e:
         raise ValidationError("Invalid supplementary data") from e
+
+
+def get_key_store() -> KeyStore:
```

Comment:
minor - do we need a method for one dict lookup?

Reply:
Shut up dumb Kyle from the past - I have removed it and just passed the key store (which can be fetched in get_supplementary_data), into the decrypt_supplementary_data method that gets called